v2/pkg/pool/limitbufferpool.go

```diff
@@ -0,0 +1,97 @@
+package pool
+
+import (
+	"bytes"
+	"context"
+	"runtime"
+	"time"
+)
+
+// LimitBufferPool is a pool of buffers that is limited in size and is limiting the max size of buffers that should be recycled
+// In addition, it runs a GC runtime that randomly resets a buffer every second to keep the memory usage low when usage is low
+// This is an alternative to sync.Pool, which can grow unbounded rather quickly
+type LimitBufferPool struct {
+	buffers []*ResolvableBuffer
+	index   chan int
+	options LimitBufferPoolOptions
+}
+
+type ResolvableBuffer struct {
+	Buf *bytes.Buffer
+	idx int
+}
+
+type LimitBufferPoolOptions struct {
+	// MaxBuffers limits the total amount of buffers that can be allocated for printing the response
+	// It's recommended to set this to the number of CPU cores available, or a multiple of it
+	// If set to 0, the number of CPU cores is used
+	MaxBuffers int
+	// MaxBufferSize limits the size of the buffer that can be recycled back into the pool
+	// If set to 0, a limit of 10MB is applied
+	// If the buffer cap exceeds this limit, a new buffer with the default size is created
+	MaxBufferSize int
+	// DefaultBufferSize is used to initialize the buffer with a default size
+	// If set to 0, a default size of 8KB is used
+	DefaultBufferSize int
+}
+
+func NewLimitBufferPool(ctx context.Context, options LimitBufferPoolOptions) *LimitBufferPool {
+	if options.MaxBufferSize == 0 {
+		options.MaxBufferSize = 1024 * 1024 * 10 // 10MB
+	}
+	if options.DefaultBufferSize < 1024*8 {
+		options.DefaultBufferSize = 1024 * 8 // 8KB
+	}
+	if options.MaxBuffers == 0 {
+		options.MaxBuffers = runtime.NumCPU()
```

Comment:
This won't be picked up in container environments. It will always use the cpu count of the hosts.